app/javascript/lib/imperial.js

```diff
@@ -1226,12 +1231,33 @@ export default class Imperial {
     } else {
       this.maxImports = 3;
     }
+
+    if (this.armyCount(action.payload.nation) == this.unitLimits.get(action.payload.nation).armies) {
+      this.availableActions = availableActions;
+      this.importing = true;
+      return;
+    }
+
+    if (this.fleetCount(action.payload.nation) == this.unitLimits.get(action.payload.nation).fleets) {
+      this.availableActions = availableActions;
+      this.importing = true;
+      return;
+    }
```

Comment:
I'm thinking this and the above section about armies can probably be deleted now. Do tests pass if we delete them? They should, I think, because the logic below should filter out any additional import actions.